app/components/HoverPreview.tsx

```diff
@@ -94,27 +118,65 @@ function HoverPreviewInternal({ element, id, onClose }: Props) {
     : anchorBounds.left;
   const leftOffset = anchorBounds.left - left;
 
-  return (
+  return !data || loading ? (
+    <LoadingIndicator />
+  ) : (
     <Portal>
       <Position
         top={anchorBounds.bottom + window.scrollY}
         left={left}
         aria-hidden
       >
         <div ref={cardRef}>
-          <HoverPreviewDocument url={element.href} id={id}>
-            {(content: React.ReactNode) =>
-              isVisible ? (
-                <Animate>
-                  <Card>
-                    <Margin />
-                    <CardContent>{content}</CardContent>
-                  </Card>
-                  <Pointer offset={leftOffset + anchorBounds.width / 2} />
-                </Animate>
-              ) : null
-            }
-          </HoverPreviewDocument>
+          {isVisible ? (
+            <Animate>
+              <Card>
+                <Margin />
+                <CardContent>
+                  {data.type === "mention" ? (
+                    <Content to="">
+                      <Flex gap={12}>
```

Comment:
Can we split these into individual components? Otherwise the `React.Suspense` will be hit immediately, and better for the near future of supporting more than 2 types here.